app/services/ems_physical_infra_dashboard_service.rb

```diff
@@ -101,36 +101,32 @@ def recent_servers
     }
   end
 
-  def notification_data(attrs)
-    valid = 0
-    warning = 0
-    critical = 0
-    attrs.each do |attr|
-      next unless attr.respond_to?(:health_state)
-      case attr.health_state&.downcase
-      when 'critical'
-        critical += 1
-      when 'warning'
-        warning += 1
-      when 'valid'
-        valid += 1
+  def notification_data(components, component_type)
+    count = 0
+    icon_class = ''
+    description = ''
+    if components&.first.respond_to?(:health_state)
+      health_states = components.group('lower(health_state)').count
+      health_states.default = 0
+      valid_count = health_states["valid"]
+      warning_count = health_states["warning"]
+      critical_count = health_states["critical"]
+
+      if critical_count.positive?
+        icon_class = 'pficon pficon-error-circle-o'
+        count = critical_count
+        description = _("#{count} #{component_type.downcase} are in critical state.")
```

Comment:
The above won't work. You need the following:

```ruby
_("%{count} %{component_type} are in critical state.") % {:count => count, :component_type => component_type.downcase}```

Same below.

Reply:
Does this thing have a singular/plural form? I see `1 servers are in critical state` as a tooltip.

Reply:
hey @mzazrivec I will change this, but could explane what won't work ?

Reply:
@skateman This doesn't have singular/plural, If you want I could add as well.

Reply:
> I got the tooltip, but the number is still there:

@skateman About that, the idea is only to remove it if all components are healthy. If we have any state as warning or critical we should show their count.

Reply:
It seems a little inconsistent for me, @epwinchell what do you think?

Reply:
@CharlleDaniel for the singular/plural, you can use `n_(singular_text, plural_text, number)`

Reply:
> hey @mzazrivec I will change this, but could you explain what won't work ?

i18n won't work.

In the following code: `var = "random"; _("#{var} string")'`, at first the inside of the gettext call is evaluated, i.e. gettext will be looking for the following string: `"random string"`. Such string though won't exist in the catalog.

In the following: `var = "random"; _("%{var} string") % {:var => var};`, at first the gettext call is evaluated and then the string expansion takes place.

Reply:
@skateman It feels a bit inconsistent, but it is consistent with the PF example: https://www.patternfly.org/pattern-library/dashboard/dashboard-layout/ So, I think this is fine.

Reply:
@skateman Now was added a singular and plural message. Could you review again?

Reply:
@mzazrivec Done, the gettext was fixed. Also thank you for your explanation, was very very nice.

Reply:
I think you should use `n_()`, if it wasn't working you probably called it wrong. Can you send me over the line and the stacktrace?

Reply:
@skateman I'm sorry, I did try to use `_(singular_text, plural_text, number) ` but now I'm using `n_()` and it's works very well. Could you review again?